auth-lib.js

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env node
+
+/**
+ * This file wraps auth-service.js and allows an authorization code to be retrieved
+ * without using the CLI.
+ */
+const authService = require("./auth-service");
+
+const defaultScope = [
+  "openid",
+  "profile",
+  "offline_access",
+  "email",
+  "address phone",
+];
+
+const defaultOptions = {
+  show: false,
+  sandbox: true,
+  scope: defaultScope,
+  pkce: false,
+  debug: false,
+  dir: "/tmp",
+  speed: 1,
+  idp: "idme",
+};
+
+/**
+ * Calls the main function in auth-service.js to retrieve an authorization code.
+ * See yargs in auth.js for more info on accepted parameters
+ */
+async function getAuthCode(clientId, clientSecret, email, password, options) {
+  options = Object.assign(defaultOptions, options);
+  const args = {
+    "client-id": clientId,
+    "client-secret": clientSecret,
+    "user-email": email,
+    "user-password": password,
+    ...options,
+  };
+
+  const authCode = await authService.main(args);
+  return authCode;
```

Comment:
Currently in some circumstances an error is being returned from auth-service.js, but that error is getting returned like a normal response object. auth-lib.js needs to check if the response is an error response, and if it is an error should be thrown

```suggestion
  const response = await authService.main(args);
  
  if (response.error) {
    throw new Error(response.detail);
  }
  
  return response;
```